chains/orchestrator-paras/runtime/dancebox/src/tests/inactivity_tracking.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+#![cfg(test)]
+
+use {
+    crate::tests::common::*,
+    frame_support::traits::Get,
+    pallet_inactivity_tracking::pallet::{ActiveCollators, ActiveCollatorsForCurrentSession},
+    tp_traits::{
+        AuthorNotingHook, AuthorNotingInfo, MaybeSelfChainBlockAuthor, NodeActivityTrackingHelper,
+        ParaId,
+    },
+};
+
+fn get_author_noting_info(
+    author: &AccountId,
+    container_chain: &ParaId,
+) -> AuthorNotingInfo<AccountId> {
+    AuthorNotingInfo {
+        block_number: System::block_number(),
+        author: author.clone(),
+        para_id: *container_chain,
+    }
+}
+
+fn note_block_authors(authors: Vec<(AccountId, ParaId)>) {
+    let mut authors_info: Vec<AuthorNotingInfo<AccountId>> = Vec::new();
+    authors.iter().for_each(|block_info| {
+        authors_info.push(get_author_noting_info(&block_info.0, &block_info.1))
+    });
+    let _ = InactivityTracking::on_container_authors_noted(&authors_info.as_slice());
+}
+
+#[test]
+fn inactivity_tracking_correctly_updates_storages() {
+    ExtBuilder::default()
+        .with_empty_parachains(vec![3000, 3001])
+        .with_collators(vec![
+            (AccountId::from(ALICE), 100_000),
+            (AccountId::from(BOB), 100_000),
+            (AccountId::from(CHARLIE), 100_000),
+            (AccountId::from(DAVE), 100_000),
+        ])
+        .build()
+        .execute_with(|| {
+            run_block();
+            note_block_authors(vec![(CHARLIE.into(), 3001.into())]);
```

Comment:
I would separate this into 2 tests:

- one where we test the author noting for the orchestrator chain. that one should be rather easy, as every run_block() we simply inject an author log

- one where we test the author noting for the container-chains. In this case, we need to use something similar to https://github.com/moondance-labs/tanssi/blob/04b18155ddb9dd85b396196a72299dfcbedfe1df/chains/orchestrator-paras/runtime/dancebox/src/tests/integration_test.rs#L4838 to inject the log. This are supposed to be integration tests, ergo we should try to go through the real path instead of directly calling it like you are doing in `note_block_authors`. we can discuss these further if you need help

Reply:
I split the dancebox  inactivity tracking test into 2 tests in https://github.com/moondance-labs/tanssi/pull/922/commits/47da736d18bcabdf3f15647d89dc5ad7bda6d36d and replaced `note_block_authors `with `ParaHeaderSproofBuilder` in https://github.com/moondance-labs/tanssi/pull/922/commits/2a829cf9b759dcecccb15f98c65ff08b41eeb7e0